src/main/java/sora/task/TaskList.java

```diff
@@ -0,0 +1,286 @@
+package sora.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import sora.exception.EmptyListException;
+import sora.exception.IllegalFormatException;
+import sora.exception.TaskNotFoundException;
+import sora.util.Message;
+
+/**
+ * Represents a list of tasks.
+ *
+ * @author Zhang Shi Chen
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Constructor for TaskList.
+     */
+    public TaskList() {
+        this(new ArrayList<>());
+    }
+
+    /**
+     * Overloading constructor for TaskList.
+     * Allows user to input a list of tasks.
+     *
+     * @param tasks an existing list of tasks
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Deletes the task with the same task number specified by user.
+     *
+     * @param command command entered by user (delete [task number])
+     * @return confirmation message for successfully deleted a task
+     * @throws IllegalFormatException if user gives empty or invalid task number
+     * @throws TaskNotFoundException if the task specified by the task number does not exist
+     */
+    public String taskDelete(String command) throws IllegalFormatException, TaskNotFoundException {
+        // Throw exception if user gives empty or non-integer task number
+        validateCommand(command, "^delete [0-9]+", "delete [task number]");
+
+        int taskNumber;
+        try {
+            taskNumber = Integer.parseInt(command.substring(7)) - 1;
+        } catch (NumberFormatException e) {
+            throw new TaskNotFoundException();
+        }
+
+        // Throw exception if taskNumber is out of range
+        if (taskNumber < 0 || taskNumber >= tasks.size()) {
+            throw new TaskNotFoundException();
+        }
+
+        // Delete the task
+        Task task = tasks.remove(taskNumber);
+
+        int size = tasks.size();
+
+        // Return confirmation message
+        return Message.DELETE + "\n  "
+            + task + "\n" + (
+            size == 0
+            ? "You have no more task in the list!"
```

Comment:
Small grammar error "You have no more task**s** in the list!", really small though

Reply:
Thx for pointing out... too long nvr write proper eng alr